### ISSUE TITLE
Type error when restoring session

### ISSUE THREAD

**Issue by DeantheDerq** [bethgelab/foolbox#100]
I am not sure if this is bug is because i messed up using fool box or because I messed up restoring but when I try to restore the checkpoint file i get various errors that switch around depending on how i try to solve the previous one.  I've been trying to follow the code (https://github.com/bethgelab/foolbox/issues/12) . I guess what I am messing up or what i don't understand is what it means by replace with restorer in issue 12. I have read the tutorial but I still don't quite understand because the context seems to be different. Any help is greatly appreciated! 

the error surrounds this
 `model.session.run( saver.restore( tf.Session(),  "/tmp/models/convnet_maxpool.ckpt" ) )`

This is the rest of the code: 

    import foolbox
    import tensorflow as tf
    import numpy as np

    inputs = tf.placeholder(tf.float32, shape=(None, 784))
    logits = tf.layers.dense(inputs, 10)

    init_op = tf.global_variables_initializer()

    np.random.seed(2)
    W = np.random.rand(784, 10).astype(np.float32)

    assign_op = tf.assign(tf.global_variables()[0], W)

    np.random.seed(22)
    example_input = np.random.rand(784)

    with foolbox.models.TensorFlowModel(inputs, logits, (0, 1)) as model:
        attack = foolbox.attacks.FGSM(model)

        saver = tf.train.import_meta_graph('/tmp/models/convnet_maxpool.ckpt.meta')

        model.session.run(init_op)
        model.session.run( saver.restore(tf.Session(),  "/tmp/models/convnet_maxpool.ckpt" ) )  # replace with restorer
        example_label = np.argmax(model.predictions(example_input))
        print(example_label)

        adversarial = attack(example_input, example_label, unpack=False)
        print(np.argmax(model.predictions(adversarial.image)))

        print(adversarial.distance)
